includes/modules/pages/index/header_php.php

```diff
@@ -13,31 +13,104 @@
+// least one product or one sub-category.
+//
 $category_depth = 'top';
+$current_category_not_found = false;
+$current_category_is_disabled = false;
+$current_category_has_products = false;
+$current_category_has_subcats = false;
 if (isset($cPath) && zen_not_null($cPath)) {
-    $categories_products_query = 
-        "SELECT count(*) AS total
-           FROM   " . TABLE_PRODUCTS_TO_CATEGORIES . "
-          WHERE   categories_id = :categoriesID";
-
-    $categories_products_query = $db->bindVars($categories_products_query, ':categoriesID', $current_category_id, 'integer');
-    $categories_products = $db->Execute($categories_products_query);
-
-    if ($categories_products->fields['total'] > 0) {
-        $category_depth = 'products'; // display products
+    $category_status_query = 
+        "SELECT categories_status
+           FROM " . TABLE_CATEGORIES . "
+          WHERE categories_id = :currentCategoryId
+          LIMIT 1";
+    $category_status_query = $db->bindVars($category_status_query, ':currentCategoryId', $current_category_id, 'integer');
+    $category_status = $db->Execute($category_status_query);
+    if ($category_status->EOF) {
+        $current_category_not_found = true;
+    } elseif ($category_status->fields['categories_status'] == '0') {
+        $current_category_is_disabled = true;
+    }
+    $category_products_query = 
+        "SELECT products_id
+           FROM " . TABLE_PRODUCTS_TO_CATEGORIES . "
+          WHERE categories_id = :currentCategoryId
+          LIMIT 1";
+    $category_products_query = $db->bindVars($category_products_query, ':currentCategoryId', $current_category_id, 'integer');
+    $category_products = $db->Execute($category_products_query);
+    if (!$category_products->EOF) {
+        $current_category_has_products = true;
     } else {
-        $category_parent_query = 
-            "SELECT count(*) AS total
-               FROM   " . TABLE_CATEGORIES . "
-              WHERE  parent_id = :categoriesID";
-
-        $category_parent_query = $db->bindVars($category_parent_query, ':categoriesID', $current_category_id, 'integer');
+        $category_parent_query =
+            "SELECT parent_id
+               FROM " . TABLE_CATEGORIES . "
+              WHERE parent_id = :currentCategoryId
+              LIMIT 1";
+        $category_parent_query = $db->bindVars($category_parent_query, ':currentCategoryId', $current_category_id, 'integer');
         $category_parent = $db->Execute($category_parent_query);
-
-        if ($category_parent->fields['total'] > 0) {
-            $category_depth = 'nested'; // navigate through the categories
+        $current_category_has_subcats = !$category_parent->EOF;
+    }
+    
+    // -----
+    // Give an observer the chance to override the default handling for the category.
+    //
+    $category_redirect_handled = false;
+    $zco_notifier->notify(
+        'NOTIFY_HEADER_INDEX_HTTP_STATUS_CHECK', 
```

Comment:
Do you mind having a discussion about naming of this?
`NOTIFY_HEADER_INDEX_HTTP_STATUS_CHECK`

Given the location in code where it fires, the name makes perfect sense. That would be my initial first thought too, especially given the product-related pattern this code was inspired from.

But given its very specialized purpose and that all the data passed to it is strictly category-related, are `INDEX` and `HTTP_STATUS_CHECK` both good namings here?

Reply:
What do you have in mind?  Something more like `NOTIFY_INDEX_CATEGORY_MISSING_DISABLED_CHECK`?

Reply:
> What do you have in mind? Something more like `NOTIFY_INDEX_CATEGORY_MISSING_DISABLED_CHECK`?

Ya, something along that line feels context-relevant. 
Or maybe `NOTIFY_INDEX_CATEGORY_STATUS_CHECK` because it's a bit shorter.

Reply:
Updated in that update just pushed.